service/db.go

```diff
@@ -0,0 +1,71 @@
+package service
+
+import (
+	"fmt"
+	"github.com/hwsc-org/hwsc-document-svc/conf"
+	"github.com/mongodb/mongo-go-driver/mongo"
+	"golang.org/x/net/context"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+var (
+	mongoDBReader *mongo.Client
+	mongoDBWriter *mongo.Client
+)
+
+func init() {
+	var err error
+	mongoDBReader, err = dialMongoDB(conf.DocumentDB.Reader)
+	if err != nil {
+		log.Fatalf("[FATAL] %s\n", err.Error())
+	}
+	mongoDBWriter, err = dialMongoDB(conf.DocumentDB.Writer)
+	if err != nil {
+		log.Fatalf("[FATAL] %s\n", err.Error())
+	}
+	// Handle Terminate Signal(Ctrl + C)
+	c := make(chan os.Signal)
+	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
+	go func() {
+		<-c
+		_ = disconnectMongoDBClient(mongoDBReader)
+		_ = disconnectMongoDBClient(mongoDBWriter)
+		fmt.Println()
+		log.Fatalln("[FATAL] hwsc-document-svc terminated")
+	}()
+}
+
+// dialMongoDB connects a client to MongoDB server.
+// Returns a MongoDB Client or any dialing error.
+func dialMongoDB(uri string) (*mongo.Client, error) {
+	return mongo.Connect(context.TODO(), uri)
+
```

Comment:
wanted to point out this extra line, idc about it, but you might

Reply:
i care